tests/unit/test_scm.py

```diff
@@ -117,3 +118,84 @@ def test_clone_as_tarball_wrong_ref(golang_repo_path: Path, tmp_path: Path) -> N
         match=f'Please verify the supplied reference of "{bad_commit}" is valid',
     ):
         clone_as_tarball(f"file://{golang_repo_path}", bad_commit, tmp_path / "my-repo.tar.gz")
+
+
+class TestGetRepoForPath:
+    """Tests for get_repo_for_path and related submodule handling."""
+
+    @pytest.mark.parametrize(
+        "path_from_main_root,expected_repo_path,expected_relative_path",
+        [
+            pytest.param("", ".", ".", id="main_repo_root"),
+            pytest.param("src", ".", "src", id="directory_in_main"),
+            pytest.param("src/utils", ".", "src/utils", id="nested_directory_in_main"),
+            pytest.param("README.md", ".", "README.md", id="file_in_main"),
+            pytest.param("submodule", "submodule", ".", id="submodule_root"),
+            pytest.param("submodule/src", "submodule", "src", id="directory_in_submodule"),
+            pytest.param(
+                "submodule/src/utils", "submodule", "src/utils", id="nested_dir_in_submodule"
+            ),
+            pytest.param(
+                "submodule/submodule_file.txt",
+                "submodule",
+                "submodule_file.txt",
+                id="file_in_submodule",
+            ),
+            pytest.param("submodule/nested", "submodule/nested", ".", id="nested_submodule_root"),
+            pytest.param(
+                "submodule/nested/nested_file.txt",
+                "submodule/nested",
+                "nested_file.txt",
+                id="file_in_nested",
+            ),
+            pytest.param(
+                "submodule/nested/deep/dir", "submodule/nested", "deep/dir", id="deep_dir_in_nested"
+            ),
+        ],
+    )
+    def test_path_resolution(
+        self,
+        path_from_main_root: str,
+        expected_repo_path: str,
+        expected_relative_path: str,
+        repo_with_nested_submodules: git.Repo,
+    ) -> None:
+        """Test path resolution across nested submodules."""
+        main_repo_root = Path(repo_with_nested_submodules.working_dir)
+
+        # Create the target path (directories or files)
+        full_target_path = main_repo_root / path_from_main_root
+        if "." not in full_target_path.name:  # No extension means it's a directory
```

Comment:
I am sorry, I feel very dense, but why are paths with dots in them neglected completely? I can see several in parametrization list.

Reply:
Dropped from the revised test